### PR TITLE
Add more commands for Gator protocol

### PR DESCRIPTION
**Pull request by seym45**
List of commands added for Gator protocol

1. positionPeriodic  (Tested)
```
"attributes": {
        "heartbeat_interval": 120,
        "engine_on_interval": 30,
        "engine_off_interval": 60
}
```
2. setSpeedLimit
```
"attributes": {
        "data": 120
}
```
3. setOdometer
```
"attributes": {
        "data": 12023
}
```
https://www.traccar.org/protocol/5052-gator/S208%20S228%20protocolENGLISH-st.doc

### CODE REVIEW

**Comment by tananaev** [comment]
Don't forget to include the protocol documentation in the description.

**Comment by seym45** [comment]
From log:
2023-08-22 05:37:40  INFO: [T4ec6b58b] id: 13088007384, command type: positionPeriodic sent
2023-08-22 05:37:40  INFO: [T4ec6b58b: gator > IP] 242434000b580049540028002878020d
2023-08-22 05:37:42  INFO: [T4ec6b58b: gator < IP] 242485000b580049543400010000fe0d

**Comment by seym45** [comment]
Set speed limit
```
2023-08-22 06:19:20  INFO: [T1f884c3d] id: 13088007384, command type: setSpeedLimit sent
2023-08-22 06:19:20  INFO: [T1f884c3d: gator > x.x.x.x] 24243f00075800495478050d
2023-08-22 06:19:22  INFO: [T1f884c3d: gator < x.x.x.x] 242485000b580049543f780100008d0d
2023-08-22 06:19:22  INFO: [T1f884c3d: gator > x.x.x.x] 24242100058d85002c0d
```

Set odometer
```
2023-08-22 06:18:13  INFO: [T1f884c3d] id: 13088007384, command type: setOdometer sent
2023-08-22 06:18:13  INFO: [T1f884c3d: gator > x.x.x.x] 24246b00085800495400785e0d
2023-08-22 06:18:16  INFO: [T1f884c3d: gator < x.x.x.x] 242480002358004954230822061734005133531002913200000047c0470100000d3b073c2485270d                                                            
2023-08-22 06:18:16  INFO: [T1f884c3d] id: 13088007384, time: 2023-08-22 14:17:34, lat: 5.22255, lon: 100.48553, course: 47.0                                                                              
2023-08-22 06:18:16  INFO: [T1f884c3d: gator > x.x.x.x] 2424210005278000830d
2023-08-22 06:18:18  INFO: [T1f884c3d: gator < x.x.x.x] 242485000b58004954210d010000e60d
2023-08-22 06:18:18  INFO: [T1f884c3d: gator > x.x.x.x] 2424210005e68500470d
```

**Comment by seym45** [comment]
Hi @tananaev,

Can you please review?

File: src/main/java/org/traccar/model/Command.java [hunk 1]
```diff
@@ -72,6 +71,10 @@ public class Command extends BaseCommand {
 
     public static final String KEY_UNIQUE_ID = "uniqueId";
     public static final String KEY_FREQUENCY = "frequency";
+
+    public static final String KEY_ENGINE_OFF_INTERVAL = "engine_off_interval";
+    public static final String KEY_ENGINE_ON_INTERVAL = "engine_on_interval";
+    public static final String KEY_HEARTBEAT_INTERVAL = "heartbeat_interval";
```
Comment (tananaev) [inline src/main/java/org/traccar/model/Command.java:77]: We don't add new commands just like this. We only add if something is used by at least a few protocol and is generic enough.

Comment (seym45) [reply src/main/java/org/traccar/model/Command.java]: Used KEY_FREQUENCY to get the params

File: src/main/java/org/traccar/protocol/GatorProtocolEncoder.java [hunk 2]
```diff
@@ -66,9 +72,27 @@ private ByteBuf encodeContent(long deviceId, int type) {
...
+                content.writeByte(command.getInteger(Command.KEY_DATA));
+                return encodeContent(command.getDeviceId(), GatorProtocolDecoder.MSG_SET_SPEED_LIMIT, content);
+            case Command.TYPE_SET_ODOMETER:
+                content.writeShort(command.getInteger(Command.KEY_DATA));
+                return encodeContent(command.getDeviceId(), GatorProtocolDecoder.MSG_SET_ODOMETER, content);
+            case Command.TYPE_POSITION_PERIODIC:
+                String[] data = command.getString(Command.KEY_FREQUENCY).split(";");
+                content.writeShort(Integer.valueOf(data[0]));
+                content.writeShort(Integer.valueOf(data[1]));
+                content.writeByte(Integer.valueOf(data[2]));
```
Comment (tananaev) [inline src/main/java/org/traccar/protocol/GatorProtocolEncoder.java:94]: We don't pass the data like this.

File: src/test/java/org/traccar/protocol/GatorProtocolEncoderTest.java [hunk 3]
```diff
@@ -20,4 +20,17 @@ public void testEncode() throws Exception {
         command.setType(Command.TYPE_POSITION_SINGLE);
         verifyCommand(encoder, command, binary("24243000062008958C070D"));
     }
+
+    @Test
+    public void testEncodePeriodicPositionRetrievalIntervalSet() throws Exception {
```
Comment (tananaev) [inline src/test/java/org/traccar/protocol/GatorProtocolEncoderTest.java:25]: Method name is way too long.

File: src/main/java/org/traccar/protocol/GatorProtocolEncoder.java (hunk 2 above)
Comment (seym45) [reply src/main/java/org/traccar/protocol/GatorProtocolEncoder.java:94]: Do you have any reference or suggestion?

Comment (tananaev) [reply src/main/java/org/traccar/protocol/GatorProtocolEncoder.java:94]: We only support a single value.

Comment (seym45) [reply src/main/java/org/traccar/protocol/GatorProtocolEncoder.java:94]: image

Gator protocol needs to pass three parameters. Can we please accommodate those?

Comment (tananaev) [reply src/main/java/org/traccar/protocol/GatorProtocolEncoder.java:94]: As I said, currently not.

**Comment by seym45** [comment]
Thank you too.

**Pull request merged**